src/charm.py

```diff
@@ -303,6 +310,13 @@ def _on_certificate_expiring(self, event: CertificateExpiringEvent) -> None:
         )
         self._stored.csr = new_csr.decode()
 
+    def _on_cert_transfer_relation_joined(self, event: RelationJoinedEvent):
+        if self.certificates.cert:
+            certificate = self.certificates.cert
+            ca = self.certificates._ca_cert
+            chain = self.certificates._chain
```

Comment:
I wonder why these are private. Perhaps raise an issue on the tls-cert lib repo?

Reply:
kinda related: we forgot it as well in certhandler. https://github.com/canonical/observability-libs/pull/54